tests/Adapters/DatabasePersistentPageIdentifiersRepoTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\PersistentPageIdentifiers\Tests\Adapters;
+
+use MediaWikiIntegrationTestCase;
+use ProfessionalWiki\PersistentPageIdentifiers\Adapters\DatabasePersistentPageIdentifiersRepo;
+use ProfessionalWiki\PersistentPageIdentifiers\Adapters\PersistentPageIdentifiersRepo;
+use Wikimedia\Rdbms\DBQueryError;
+
+/**
+ * @covers \ProfessionalWiki\PersistentPageIdentifiers\Adapters\DatabasePersistentPageIdentifiersRepo
+ * @group Database
+ */
+class DatabasePersistentPageIdentifiersRepoTest extends MediaWikiIntegrationTestCase {
+
+	protected function setUp(): void {
+		parent::setUp();
+		$this->tablesUsed[] = 'persistent_page_ids';
+	}
+
+	private function newRepo(): PersistentPageIdentifiersRepo {
+		return new DatabasePersistentPageIdentifiersRepo( $this->db );
+	}
+
+	public function testGetPersistentIdReturnsNullForPageWithoutPersistentId(): void {
+		$repo = $this->newRepo();
+
+		$this->assertNull( $repo->getPersistentId( 404 ) );
+	}
+
+	public function testSetPersistentIdCanBeRetrievedWithGetPersistentId(): void {
```

Comment:
I think the `WithGetPersistentId` part at the end is superfluous

Reply:
I'd have done something like `testCanSaveAndRetrievePersistendId` or `testCanRetrieveSavedPersitendId`